### ISSUE TITLE
[Feature]: SubServers 2 Intergreation

### ISSUE THREAD

**Issue by ajh123** [dieterblancke/BungeeUtilisalsX#221]
# Introduction
[SubServers 2](https://github.com/ME1312/SubServers-2) is a Velocity / Bungeecord server management system. It allows server owners to dynamically create, delete, and update backend severs (Bukitt / Spigot / Paper / Forge / Fabric) at runtime without restarting the proxy. Also, SubServers 2 have the ability to run on multiple proxies at once.

# What could be done
* Support for automatically finding groups by using SubServers 2 groups.
* With commands (like [/server](https://docs.endoy.dev/BungeeUtilisalsX/#/commands?id=server)) allow [SubServers 2 referencing](https://github.com/ME1312/SubServers-2/wiki/Commands-and-Permissions#selecting-multiple-servers).
* Any thing else you could think of.

# Additional information
See https://github.com/ME1312/SubServers-2/wiki/SubAPI for API details.

**Comment by dieterblancke** [comment]
Correct me if I'm wrong, but lets say you have a sub servers group "Test", does that then create servers like "Test-1", "Test-2", ...?

If so, wouldn't it be sufficient if you used the servergroups system that comes included with BungeeUtilisalsX?

**Comment by ajh123** [comment]
You have to create the servers first and add them in the group inside the `servers.yml`.

It would be sufficient to use built in servergroups however it would get complicated because you would have to do it twice, one for SubServers 2 and BungeeUtilisalsX. Also SubServers 2 can add servers / groups while the proxy is running (like Cloud Net).

This could work with [Dynamic ServerGroups](https://docs.endoy.dev/BungeeUtilisalsX/#/other?id=dynamic-servergroups).

**Comment by dieterblancke** [comment]
> You have to create the servers first and add them in the group inside the `servers.yml`.
> 
> It would be sufficient to use built in servergroups however it would get complicated because you would have to do it twice, one for SubServers 2 and BungeeUtilisalsX. Also SubServers 2 can add servers / groups while the proxy is running (like Cloud Net).
> 
> This could work with [Dynamic ServerGroups](https://docs.endoy.dev/BungeeUtilisalsX/#/other?id=dynamic-servergroups).

Yeah, that's what I'm wondering about, is it sufficient with (dynamic) server groups?

Or otherwise, the feature you're asking is to add SubServers2 support to fetch groups from their api on startup?

**Comment by ajh123** [comment]
> Or otherwise, the feature you're asking is to add SubServers2 support to fetch groups from their api on startup?

Basically that but fully dynamic. You could use the [SubAddServerEvent](https://dev.me1312.net/jenkins/job/SubServers%20Platform/javadoc/SubServers.Sync.Velocity/net/ME1312/SubServers/Velocity/Event/SubAddServerEvent.html) then lookup the server's group later.

**Comment by dieterblancke** [comment]
The ServerGroup system currently supports dynamic servers, so it can be used to support SubServers 2.
